output/transform/merge_partition_sizes.go

```diff
@@ -12,11 +12,17 @@ import (
 //
 // TODO: recursively build up stats when nested partitioning is used
 func mergePartitionSizes(s snapshot.FullSnapshot, newState state.PersistedState, ts state.TransientState, databaseOidToIdx OidToIdx) snapshot.FullSnapshot {
+	relIdxToStatsIdx := make(map[int32]int, len(s.RelationStatistics))
+	for idx, stat := range s.RelationStatistics {
```

Comment:
Should we add the same check as lines 21-23 below here?

Reply:
Whilst that'd keep the map smaller, I don't think its worth the effort (either solution should be correct).

Reply:
Right, that's what I was thinking about. Okay, thanks.